snuba/consumer.py

```diff
@@ -48,9 +48,12 @@ def __init__(
     def process_message(
         self, message: Message[KafkaPayload]
     ) -> Optional[ProcessedMessage]:
-        # TODO: consider moving this inside the processor so we can do a quick
-        # processing of messages we want to filter out without fully parsing the
-        # json.
+        pre_filter = (
+            enforce_table_writer(self.__dataset).get_stream_loader().get_pre_filter()
+        )
+        if pre_filter and pre_filter.should_drop(message):
+            return None
+
         if self.__rapidjson_deserialize:
```

Comment:
I did not make the parsing phase storage specific as well because, as of now, none needs that, though we could easily go that way and actually make the storage provide all the components used by this class to process a message.

Reply:
This comment was made prior to the discussion of entities — my understanding is approach wouldn't make as much sense under those circumstances, since the consumer will be transforming messages to entities, which are then transformed to storage rows/operations based on the writable storage implementation?

Reply:
I think the consumer will change even more with entities. Yes your pipeline seems the one we discussed. I think a concept of pre-filtering and then parsing can still be valid there even though it would look different.